### PR TITLE
My Posts Added

### LINKED ISSUE #44: View Current User's Posts
As an author, I would like to see a list of all the Posts I have written so that I can easily view, edit, delete, publish or unpublish them.

**Given** the user is in the Tabloid application  
**When** they select the `My Posts` menu option  
**Then** they should be directed to the "My Posts" list page  
**And** the page should display ALL the Posts authored by the logged-in user  
**And** each post in the list should display the title, author and category  
**And** the list should be in order of creation date with the most recent on top

### PR DESCRIPTION
**Pull request by SchmidtFred**
Resolves #44 

#### Changes Made
1. Added file `MyPosts` to `Components/Posts` directory.
1. Modified file `PostRepository` and `IPostRepository` to include `GetPostsByUser`, `GetPostsByCurrentUser`, and functionality.
2. Modified file `PostController` to include `GetPostsByUser`, `GetPostsByCurrentUser`, `GetCurrentUserProfile`, and functionality.
3. Modified file `PostManager` to include `GetAllPostsByCurrentUser` and functionality.
4. Modified file `MyPosts` to include rendering functionality.
5. Modified file `ApplicationViews` to include MyPosts view.
6. Modified file `Header` to include My Posts option.

#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout fs-CurrentUserPosts
    ```
7. Open a new Terminal tab (⌘T) and navigate to the server directory.
8. Test app functionality.
    > Given a user is logged in, When the user clicks the "My Posts" option in the header, Then they are presented with a list of all posts they have created sorted by recent.
9. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

**Review by matt-rich-anderson — APPROVED** [review]
Feature works as expected. I reviewed the code and nothing stood out to me. You have my approval.

File: Tabloid/Controllers/PostController.cs [hunk 1]
```diff
@@ -34,5 +37,29 @@ public IActionResult Get(int id)
             }
             return Ok(post);
         }
+
+        [HttpGet("GetPostsByUser/{id}")]
```
Comment (joshdbarton) [inline Tabloid/Controllers/PostController.cs]: I would prefer to not see verbs in the resource names in the URLs. you've already got an HTTP verb, so PostsByUser is sufficient (we'll talk more about this today in the talk about REST)

File: Tabloid/Controllers/PostController.cs [hunk 2]
```diff
@@ -34,5 +37,29 @@ public IActionResult Get(int id)
...
+        public IActionResult GetPostsByUser(int id)
+        {
+            return Ok(_postRepo.GetAllPostsByUser(id));
+        }
+
+        [HttpGet("GetPostsByCurrentUser")]
```
Comment (joshdbarton) [inline Tabloid/Controllers/PostController.cs]: same here

Comment (joshdbarton) [reply Tabloid/Controllers/PostController.cs]: Maybe MyPosts?

File: Tabloid/Repositories/PostRepository.cs [hunk 3]
```diff
@@ -50,6 +50,48 @@ public List<Post> GetAllPublishedPosts()
...
+                               p.CreateDateTime AS PostCreateDateTime,
+                               p.PublishDateTime, p.IsApproved,
+                               p.CategoryId, p.UserProfileId,
+
+                               u.FirstName, u.LastName, u.DisplayName,
+                               u.Email, u.CreateDateTime AS UserCreateDateTime, u.ImageLocation AS AvatarImage,
+                               u.UserTypeId,
+                           
+                               ut.[Name] AS UserTypeName
+                          FROM Post p
```
Comment (joshdbarton) [inline Tabloid/Repositories/PostRepository.cs:66]: I get that you might need all of this data to satisfy the NewPostFromReader method. I'm wondering whether you might be returning more data than you need back to the requesting client. Maybe you do need it all, just asking you to think about it.

File: Tabloid/client/src/modules/postManager.js [hunk 4]
```diff
@@ -33,4 +33,21 @@ export const getPost = (id) => {
...
+        return fetch(`${baseUrl}/GetPostsByCurrentUser`, {
+            method: "GET",
+            headers: {
+                Authorization: `Bearer ${token}`
+            }
+        }).then(res => {
+            if (res.ok) {
+                return res.json();
+            } else {
+                throw new Error("An unkown erro occurred whilte trying to get posts.");
```
Comment (joshdbarton) [inline Tabloid/client/src/modules/postManager.js]: typo here.

**Review by joshdbarton — CHANGES_REQUESTED** [review]

File: Tabloid/Repositories/PostRepository.cs (hunk 3 above)
Comment (SchmidtFred) [reply Tabloid/Repositories/PostRepository.cs:66]: Resolved by having two PostFromReader methods, one to return with users, the other to return without them.

File: Tabloid/client/src/modules/postManager.js (hunk 4 above)
Comment (SchmidtFred) [reply Tabloid/client/src/modules/postManager.js]: Fixed.

File: Tabloid/Controllers/PostController.cs (hunk 1 above)
Comment (SchmidtFred) [reply Tabloid/Controllers/PostController.cs]: Changed to PostsByUser

File: Tabloid/Controllers/PostController.cs (hunk 2 above)
Comment (SchmidtFred) [reply Tabloid/Controllers/PostController.cs]: Changed to MyPosts

**Review by joshdbarton — APPROVED** [review]

**Pull request merged**